### ISSUE TITLE
Fold regions are weird in python-mode

### ISSUE THREAD

**Issue by ipsod** [syl20bnr/spacemacs#5482]
```
class Test():
    def whatever():
        test = 'test'
        test = 'test'
```

In Vim, using the code above, putting the cursor anywhere on the line `def whatever():` and hitting zc will close the method.  In Spacemacs `python-mode`, hitting `zc` with the cursor anywhere on or before the symbol `def` will close the class, instead of the method.

#### System Info
- OS: gnu/linux
- Emacs: 24.5.1
- Spacemacs: 0.105.14
- Spacemacs branch: develop (rev. d52eb41)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(emacs-lisp git markdown org spell-checking unimpaired python html php django version-control auto-completion syntax-checking evil-snipe search-engine gnus
            (shell :variables shell-default-height 30 shell-default-position 'bottom))
```

**Comment by syl20bnr** [comment]
I'm in the impression to have already read this :-)

**Comment by TheBB** [comment]
Fixed with origami.

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
